CHAOS-4388 HCE onboard CLI (#6199)
harness/developer-hub: docs/chaos-engineering/api-reference/hce-onboarding-api.md

---
title: HCE Onboard CLI
sidebar_position: 2
description: Describes the HCE onboard CLI and its usage.

---
This section describes the Harness Chaos Engineering (HCE) onboard CLI and its usage.

## What is the HCE onboard CLI?

The `onboard_hce_aws` CLI streamlines and simplifies the HCE onboarding process. This tool also seamlessly integrates with your AWS account and executes HCE experiments. This CLI performs other tasks that streamline the onboarding process, such as:

> 

Reviewer: Recommend using bullets instead of a numbered list here.

> 1. **ChaosInfra Setup:** The CLI installs the chaos infrastructure in the given namespace of your cluster using Harness APIs and Kubernetes permissions. After installation, it tests the activation of the infrastructure for a given timeout (default is 180 s).

Reviewer: ```suggestion
- **ChaosInfra Setup:** The CLI installs the chaos infrastructure in the given namespace of your cluster using Harness APIs and Kubernetes permissions. After installation, it tests the activation of the infrastructure for a given timeout (default is 180 s).
```

> 2. **Add OIDC Provider:** The CLI can be used to add the OIDC provider in the target account using the AWS credentials. If the given provider already exists, the CLI issues a warning and skips this step.

Reviewer: ```suggestion
- **Add OIDC Provider:** The CLI can be used to add the OIDC provider to the target account using the AWS credentials. If the given provider already exists, the CLI issues a warning and skips this step.
```

> 3. **AWS Roles:** Simplifies configuring HCE with AWS. If you wish to create a dedicated role for HCE, the CLI defines the required policies and permissions directly as CLI flags. Alternatively, if you already have a role, you can provide it as an input, and that role gets attached to the provider that you added previously. In addition, this CLI doesn't depend on other CLIs such as `aws` or `kubectl`.

Reviewer: ```suggestion
- **AWS Roles:** Simplifies configuring HCE with AWS. If you wish to create a dedicated role for HCE, the CLI defines the required policies and permissions directly as CLI flags. Alternatively, if you already have a role, you can provide it as an input, and that role gets attached to the provider that you added previously. In addition, this CLI doesn't depend on other CLIs such as `aws` or `kubectl`.
```

> 4. **Annotate Service Account:** The CLI will annotate the experiment service account on the cluster with AWS roleARN after all the configuration is done.

Reviewer: ```suggestion
- **Annotate Service Account:** The CLI will annotate the experiment service account on the cluster with AWS roleARN after all the configuration is done.
```

> 5. **Compatibility:** It provides compatibility across various Windows and Linux versions.

Reviewer: ```suggestion
- **Compatibility:** It provides compatibility across various Windows and Linux versions.
```

> 6. **Flag support:** It supports a variety of flags thereby facilitating customization option for every step.

Reviewer: ```suggestion
- **Flag support:** It supports a variety of flags thereby facilitating customization options for every step.
```

> ## How to install the onboard HCE AWS?

Reviewer: ```suggestion
## Install the onboard HCE AWS
```

Onboard HCE AWS offers pre-compiled binaries that are available for download on our [releases page](https://github.com/harness/onboard_hce_aws/releases).

To install, follow these steps.

> 1. Download the appropriate binary for your platform from the **"Assets"** section.

Reviewer: ```suggestion
1. Download the appropriate binary for your platform from the **Assets** section.
```

2. Rename the downloaded file to `onboard_hce_aws` (or `onboard_hce_aws.exe` for Windows).
3. Move this file to your `$PATH` at your preferred binary installation directory.

> 

Reviewer: Recommend adding tabs for the OS options. Added comments to remove colons if you keep these as headings and choose not to use tabs.

> ### For Linux:

Reviewer: ```suggestion
#### Linux
```

* For AMD64 / x86_64:

```bash
[ $(uname -m) = x86_64 ] && curl -Lo ./onboard_hce_aws https://github.com/harness/onboard_hce_aws/releases/download/0.0.3/cli-linux-amd64
```

* For Arm64:

```bash
[ $(uname -m) = aarch64 ] && curl -Lo ./onboard_hce_aws https://github.com/harness/onboard_hce_aws/releases/download/0.0.3/cli-linux-arm64
```

* After downloading, add the execution permissions and move it to your binary installation directory using the following commands:

```bash
chmod +x ./onboard_hce_aws
sudo mv ./onboard_hce_aws /usr/local/bin/onboard_hce_aws
```

> ### For MacOS:

Reviewer: ```suggestion
#### MacOS
```
